pkgs/games/steam/fhsenv.nix

```diff
@@ -47,7 +47,10 @@ let
   ldPath = lib.optionals stdenv.is64bit [ "/lib64" ]
   ++ [ "/lib32" ]
   ++ map (x: "/steamrt/${steam-runtime-wrapped.arch}/" + x) steam-runtime-wrapped.libs
-  ++ lib.optionals (steam-runtime-wrapped-i686 != null) (map (x: "/steamrt/${steam-runtime-wrapped-i686.arch}/" + x) steam-runtime-wrapped-i686.libs);
+  ++ lib.optionals (steam-runtime-wrapped-i686 != null) (map (x: "/steamrt/${steam-runtime-wrapped-i686.arch}/" + x) steam-runtime-wrapped-i686.libs)
+  # allows for steam play and other valve utilities to work
+  ++ lib.optionals stdenv.is64bit [ "\${HOME}/.steam/steam/ubuntu12_64" ]
+  ++ [ "\${HOME}/.steam/steam/ubuntu12_32" ];
```

Comment:
It is not meant to be necessary to add these directories to the `LD_LIBRARY_PATH` before invoking Steam, and it might even break some games if you do. The parts of Steam that need these directories in their search path are meant to edit the `LD_LIBRARY_PATH` themselves. For example, Steam adds `ubuntu12_32` and `ubuntu12_32/panorama` to the `LD_LIBRARY_PATH` *for itself*, but then removes them from the `LD_LIBRARY_PATH` that it presents to games, so that games are not all forced to use Steam's bundled version of SDL2.

Reply:
Yea, I'm trying to come up with a ld cache solution... just can only work on this in my free time :/